src/components/map/Map.tsx

```diff
@@ -157,141 +179,256 @@ export const Map = ({
     };
   }
 
+  const mapLocations = locations.map((location, index) => ({
+    ...location,
+    type: 'Feature',
+    properties: { cluster: false, id: location.id, index },
+    geometry: {
+      type: 'Point',
+      coordinates: [location.position.longitude, location.position.latitude]
+    }
+  }));
+
+  useEffect(() => {
+    if (clusteringEnabled) {
+      const index = new Supercluster({
+        radius: clusterDistance,
+        maxZoom: MAX_ZOOM_LEVEL
+      });
+      const bounds = [2, 46, 18, 56];
+      const points = mapLocations;
+
+      index.load(points);
+      superClusterRef.current = index; // Store the Supercluster instance in the ref
+
+      setClusters(index.getClusters(bounds, zoomLevel));
+    } else {
+      setClusters(mapLocations);
+    }
+
+    if (locations.length > 0 && isInitialFit) {
+      const coordinates = locations.map((loc) => [loc.position.longitude, loc.position.latitude]);
+      if (locations.length === 1) {
+        if (cameraRef.current) {
+          cameraRef.current.setCamera({
+            animationDuration: 0,
+            centerCoordinate: coordinates[0],
+            zoomLevel: ONE_MARKER_ZOOM_LEVEL
+          });
+          setIsLoading(false);
+          setIsInitialFit(false);
+        }
+      } else if (coordinates && coordinates.length > 0) {
+        const { minLng, minLat, maxLng, maxLat, deltaLng, deltaLat } = calculateBounds(coordinates);
+
+        if (cameraRef.current) {
+          cameraRef.current.fitBounds(
+            [minLng - deltaLng, minLat - deltaLat],
+            [maxLng + deltaLng, maxLat + deltaLat],
+            0
+          );
+
+          if (clusteringEnabled) {
+            setIsLoading(false);
+            setIsInitialFit(false);
+          } else {
+            setIsLoading(false);
+            setIsInitialFit(false);
+          }
```

Comment:
ääh, no difference here? if this is intended, then please remove the if-else, otherwise correct the values.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1154/commits/800969674da23ee9825df39f73ddb943008dbe34